aiohttp/http_writer.py

```diff
@@ -95,7 +95,7 @@ def write_headers(self, status_line, headers, SEP=': ', END='\r\n'):
         """Write request/response status and headers."""
         # status + headers
         headers = status_line + ''.join(
-            [k + SEP + v + END for k, v in headers.items()])
+            [k + SEP + v + END for k, v in headers.items() if v is not None])
```

Comment:
"eating" some headers is misleading for the end-user. This must fail loudly. Better would be raising exception with a sane problem description.

Reply:
It's already done implicitly when `''` is passed as a header value. This was my reasoning (as mentioned in the related issue). 

I'm open to changing it to raise an error by try caching it but I think that:

a) The same should be done for `''` values, which is trickier
b) It's a bit tricky to catch the `TypeError` and raise, since `TypeError` can be many things.

But considering the above, feel free to tell me what you prefer and I'll change it :)